### ISSUE TITLE
False positive from not-a-function when using reader to read a record

### ISSUE THREAD

**Issue by lassemaatta** [clj-kondo/clj-kondo#2266]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

`clj-kondo v2023.12.15`

**platform**

Native.

**problem**

I noticed a false positive in our code where we use the clojure reader to initialize a record. In this case we initialized one of the fields to the list `("some text")`. `clj-kondo` complained that a string is not a valid function, which is correct, but in this case the list is not evaluated.

For the record, the Clojure reference says (emphasis mine):

> deftype/defrecord can be written with a special reader syntax #my.thing[1 2 3] where:
> * each element in the vector form is passed to the deftype/defrecord’s constructor **un-evaluated**
> * the deftype/defrecord name must be fully qualified
> * only available in Clojure versions later than 1.3

**repro**

`clj-kondo --lint - <<< '(defrecord Foo [field]) #user.Foo{:field ("asd")}'`

produces

```
<stdin>:1:42: error: a string is not a function
linting took 41ms, errors: 1, warnings: 0
```

**expected behavior**

No false positive is produced.

**Comment by borkdude** [comment]
ok, will fix. perhaps you could use a vector for now?

**Comment by lassemaatta** [comment]
No worries, I can easily work around this. I wrote this ticket mostly to record (no pun intended) this behaviour, as I had not seen such record syntax before :)

**Comment by borkdude** [comment]
thanks!
